Cargo.toml

```diff
@@ -19,7 +19,8 @@ maintenance = { status = "experimental" }
 
 [dependencies]
 futures-task = "0.3"
+futures-core = "0.3"
```

Comment:
I'd like to see a feature to specifically opt into this for those who may wish to avoid a dependency on `futures-core`.

Reply:
* added a `stream` feature, disabled by default
* noted about it in the crate-level comment
* enabled `doc_auto_cfg` for `docs.rs` to show a badge next to a `Stream` instance
* fixed a test to actually use `Stream::poll_next`
* added a trivial test to check that `size_hint` is propagated, mostly for coverage